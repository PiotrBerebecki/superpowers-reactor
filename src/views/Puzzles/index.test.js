/**
 * @jest-environment jsdom
 */

import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';

import { Puzzles, mapStateToProps, mapDispatchToProps } from './index';

describe('Puzzles', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Puzzles puzzles={[{ title: 'testTitle' }]} />);
    expect(shallowToJson(wrapper)).toMatchStyledComponentsSnapshot();
  });

  it('calls componentDidMount', () => {
    const spy = jest.spyOn(Puzzles.prototype, 'componentDidMount');
    const wrapper = mount(
      <Puzzles
        puzzles={[{ title: 'test01' }]}
        getPuzzles={() => {}}
        match={{ path: 'testPath' }}
      />
    );
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });

  it('calls componentWillReceiveProps', () => {
    const spy = jest.spyOn(Puzzles.prototype, 'componentWillReceiveProps');
    const wrapper = mount(
      <Puzzles puzzles={[]} category="sameCategory" getPuzzles={() => {}} />
    );
    wrapper.instance().componentWillReceiveProps({ category: 'sameCategory' });
    expect(spy).toHaveBeenCalled();
  });

  it('calls componentWillReceiveProps', () => {
    const spy = jest.spyOn(Puzzles.prototype, 'componentWillReceiveProps');
    const wrapper = mount(
      <Puzzles puzzles={[]} category="someCategory" getPuzzles={() => {}} />
    );
    wrapper
      .instance()
      .componentWillReceiveProps({ category: 'differentCategory' });
    expect(spy).toHaveBeenCalled();
  });
});

describe('mapStateToProps', () => {
  it('returns correct portion of the state', () => {
    const state = { puzzles: [], users: [] };
    const actual = mapStateToProps(state);
    expect(actual).toEqual({ puzzles: state.puzzles });
  });
});

describe('mapDispatchToProps', () => {
  it('returns an object', () => {
    const actual = mapDispatchToProps();
    expect(typeof actual).toEqual('object');
  });
});
