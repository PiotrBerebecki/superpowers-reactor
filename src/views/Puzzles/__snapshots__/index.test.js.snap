// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puzzles renders without crashing 1`] = `
<Main>
  <styled.ul>
    <styled.li>
      testTitle
    </styled.li>
  </styled.ul>
</Main>
`;
